cpp/src/arrow/adapters/orc/adapter_options.h

```diff
@@ -0,0 +1,228 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <set>
+#include <sstream>
+
+#include "arrow/adapters/orc/adapter_common.h"
+#include "arrow/io/interfaces.h"
+#include "arrow/util/visibility.h"
+#include "orc/OrcFile.hh"
+
+namespace liborc = orc;
+
+namespace arrow {
+
+namespace adapters {
+
+namespace orc {
+// Components of ORC Writer Options
+
+/**
+ * Options for creating a Writer.
+ */
+class ARROW_EXPORT WriteOptions {
+ public:
+  WriteOptions();
+  WriteOptions(const WriteOptions&);
+  WriteOptions(WriteOptions&);
+  WriteOptions& operator=(const WriteOptions&);
+  virtual ~WriteOptions();
+
+  /**
+   * Get the ORC writer options
+   * @return The ORC writer options this WriteOption encapsulates
+   */
+  std::shared_ptr<liborc::WriterOptions> GetOrcWriterOptions() const {
+    return orc_writer_options_;
+  }
+
+  /**
+   * Set the batch size.
+   */
+  WriteOptions& SetBatchSize(uint64_t size);
+
+  /**
+   * Get the batch size.
+   * @return if not set, return default value.
+   */
+  uint64_t GetBatchSize() const;
```

Comment:
Really, instead of writing and maintaining lots of trivial accessors, I think we should use another idiom where the options are a simple `struct` and the construction of `liborc::WriterOptions` becomes private.

So this would be, for example:
```c++
struct WriteOptions {
  int64_t batch_size = ...;
  int64_t stripe_size = ...;
};

```

and the `GetOrcWriterOptions` functionality wouldn't be exposed publicly.

Reply:
I renamed it `MakeOrcWriterOptions`. I wonder whether we even need an `options.cpp` since this function fits into `adapter.cpp` better than anywhere else.